e2e_tests/provider_cfg/all/Dockerfile

```diff
@@ -17,6 +17,19 @@ RUN cd mbed-crypto-mbedcrypto-2.0.0 \
 	&& make SHARED=0
 
 WORKDIR /tmp
+# Download and install SoftHSM2
+RUN git clone https://github.com/opendnssec/SoftHSMv2.git \
+	&& cd SoftHSMv2 \
+	&& git reset --hard 20a53bd083a6134ce2230f80edda5dc8be0366bd
+
+RUN cd SoftHSMv2 \
+	&& sh autogen.sh \
+	|| sh autogen.sh
```

Comment:
Would be nice to link with the bug description as a comment here for an external eye to understand why this was needed

Reply:
Done